### PR TITLE
Add support for showing logging output.

### PR DESCRIPTION
**Pull request by dhylands**
We should probably change all of the println's to logging macros,
but I'll leave that for another PR.

r? @julienw

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -4,5 +4,10 @@ version = "0.1.0"
 authors = ["Julien Wajsberg <felash@gmail.com>"]
 
 [dependencies]
-"openzwave-stateful" = { git = "https://github.com/fxbox/openzwave-stateful-rust" }
+env_logger = "0.3.2"
+libc = "0.2.7"
```
Comment (julienw) [inline Cargo.toml:8]: nit: enough to just use 0.2 ?

File: src/main.rs [hunk 2]
```diff
@@ -19,8 +28,64 @@ fn spawn_notification_thread(rx: mpsc::Receiver<ZWaveNotification>) {
...
+            )
+        };
+        builder.format(format).filter(None, LogLevelFilter::Info);
+    }
+
+    if env::var("RUST_LOG").is_ok() {
+       builder.parse(&env::var("RUST_LOG").unwrap());
+    }
+    builder.init().unwrap();
+
```
Comment (julienw) [inline src/main.rs:88]: nit: please put this in a separate function

File: src/main.rs [hunk 3]
```diff
@@ -19,8 +28,64 @@ fn spawn_notification_thread(rx: mpsc::Receiver<ZWaveNotification>) {
...
+    } else {
+        // Plain output formatter
+        let format = |record: &LogRecord| {
+            let t = time::now();
+            format!("{}.{:03} {}{:5} {}",
+                time::strftime("%Y-%m-%d %H:%M:%S", &t).unwrap(),
+                t.tm_nsec / 1_000_000,
+                tid_str(),
+                record.level(),
+                record.args()
```
Comment (julienw) [inline src/main.rs:78]: nit: please add the location as well. See my PR for foxbox: https://github.com/fxbox/foxbox/pull/518
(same for the first if/else block of course)

File: Cargo.toml [hunk 4]
```diff
@@ -4,5 +4,10 @@ version = "0.1.0"
...
+log = "0.3"
+nix = { git = "https://github.com/nix-rust/nix.git", rev = "138080" } # Until 0.5.1 is released
+openzwave-stateful = { git = "https://github.com/fxbox/openzwave-stateful-rust" }
+time = "0.1"
```
Comment (julienw) [inline Cargo.toml:12]: TBH I'm not sure I'm happy with so many dependencies for this simple app; but I don't really mind. What do you think ?

File: src/main.rs [hunk 5]
```diff
@@ -1,8 +1,17 @@
+extern crate env_logger;
+extern crate libc;
+#[macro_use]
+extern crate log;
+extern crate nix;
```
Comment (julienw) [inline src/main.rs:5]: should you add a cfg command for the `extern crate nix` as well ? as well as adding it in a target-specific part in Cargo.toml ? http://doc.crates.io/specifying-dependencies.html#platform-specific-dependencies

Comment (dhylands) [inline src/main.rs:5]: nix is available for all of the major platforms, so I didn't think it was necessary to add a cfg.

File: src/main.rs (hunk 3 above)
Comment (dhylands) [inline src/main.rs:78]: I definitely see the advantage of showing the location, but I find it quite distracting to the flow of the logging output to have to see it all of the time. It's widely variable in width and consequently changes the indentation of the logging output quite a bit. So I'd prefer to see it turned on with some type of runtime control/config option.

File: Cargo.toml (hunk 1 above)
Comment (dhylands) [inline Cargo.toml:8]: Technically, It needs to be 0.2.8 which is when SYS_gettid was added.

File: Cargo.toml (hunk 4 above)
Comment (dhylands) [inline Cargo.toml:12]: It uses all of the dependencies, as would any real-world app, so I'm not sure why having lots of dependencies is a problem. I guess in C this is all bundled under libc. rust chose to take a finer granularity, which makes sense for cross platform stuff.

File: Cargo.toml (hunk 1 above)
Comment (julienw) [inline Cargo.toml:8]: ah yeah, in that case 0.2.8 :)

**Comment by dhylands** [comment]
Closing old PRs

**Pull request closed without merging**